external/llvm-project/mlir/include/mlir/Dialect/LLVMIR/ROCDLOps.td

```diff
@@ -247,6 +247,30 @@ def ROCDL_BarrierOp : ROCDL_Op<"barrier"> {
   let assemblyFormat = "attr-dict";
 }
 
+def ROCDL_BarrierSignalOp : ROCDL_IntrOp<"s.barrier.signal", [], [], [], 0>,
+  Arguments<(ins I32Attr:$id)> {
+  let results = (outs);
+  let assemblyFormat = "$id attr-dict";
+  string llvmBuilder =
+    "createIntrinsicCall(builder, llvm::Intrinsic::amdgcn_s_barrier_signal,builder.getInt32(op.getId()));";
+}
+
+def ROCDL_BarrierWaitOp : ROCDL_IntrOp<"s.barrier.wait", [], [], [], 0>,
```

Comment:
I'm going to note that `LLVM_IntrOpBase` from upstream has `immArgPositions`, which could be used to avoid constructing a builder yourself and allowing import from LLVM as well.

Reply:
Nice tip, thanks!